src/main/java/uk/gov/hmcts/ccd/data/casedataaccesscontrol/DefaultRoleAssignmentRepository.java

```diff
@@ -208,6 +219,39 @@ public RoleAssignmentResponse findRoleAssignmentsByCasesAndUsers(List<String> ca
         }
     }
 
+    private RoleAssignmentResponse findRoleAssignmentsByCasesAndUsers(RoleAssignmentQuery roleAssignmentQuery) {
+        int pageNumber = ROLE_ASSIGNMENT_STARTING_PAGE_NUMBER;
+        int pageSize = Integer.parseInt(applicationParams.getRoleAssignmentPageSize());
+        int totalRecords = 0;
+
+        HttpHeaders headers = securityUtils.authorizationHeaders();
+        headers.add(ROLE_ASSIGNMENT_PAGE_SIZE_HEADER, Integer.toString(pageSize));
+        RoleAssignmentResponse roleAssignmentResponse = new RoleAssignmentResponse();
+
+        do {
+            headers.put(ROLE_ASSIGNMENT_PAGE_NUMBER_HEADER, Collections.singletonList(Integer.toString(pageNumber)));
+            final var requestEntity = new HttpEntity(roleAssignmentQuery, headers);
+
+            final ResponseEntity<RoleAssignmentResponse> responseEntity = restTemplate.exchange(
+                applicationParams.amQueryRoleAssignmentsURL(),
+                HttpMethod.POST,
+                requestEntity,
+                RoleAssignmentResponse.class);
+
+            if (pageNumber == ROLE_ASSIGNMENT_STARTING_PAGE_NUMBER) {
+                totalRecords = Integer.parseInt(Objects.requireNonNull(
+                    responseEntity.getHeaders().get(ROLE_ASSIGNMENT_TOTAL_RECORDS_HEADER)).get(0));
+                roleAssignmentResponse.setRoleAssignments(new ArrayList<>(totalRecords));
+            }
+
+            List<RoleAssignmentResource> roleAssignments = roleAssignmentResponse.getRoleAssignments();
```

Comment:
a minor point, but could be better to populate into a local instance and set it at the end.